public/app/features/alerting/unified/components/silences/MatchedSilencedRules.tsx

```diff
@@ -1,110 +1,81 @@
-            .flatMap((rule) => findAlertInstancesWithMatchers(rule.alerts ?? [], matchers));
-        });
-      });
-      setMatchedAlertRules(matchedInstances);
-    },
-    500,
-    [combinedNamespaces, matchers]
+  const { currentData: alerts = [] } = useGetAlertmanagerAlertsQuery(
+    { amSourceName, filter: { matchers } },
+    { skip: matchers.length === 0 }
   );
 
+  const tableItemAlerts = alerts.map<DynamicTableItemProps<AlertmanagerAlert>>((alert) => ({
+    id: alert.fingerprint,
+    data: alert,
+  }));
+
   return (
     <div>
       <h4 className={styles.title}>
         Affected alert instances
-        {matchedAlertRules.length > 0 ? (
-          <Badge className={styles.badge} color="blue" text={matchedAlertRules.length} />
+        {tableItemAlerts.length > 0 ? (
+          <Badge className={styles.badge} color="blue" text={tableItemAlerts.length} />
         ) : null}
       </h4>
       <div className={styles.table}>
         {matchers.every((matcher) => !matcher.value && !matcher.name) ? (
           <span>Add a valid matcher to see affected alerts</span>
         ) : (
-          <DynamicTable
-            items={matchedAlertRules}
-            isExpandable={false}
-            cols={columns}
-            pagination={{ itemsPerPage: 5 }}
-          />
+          <DynamicTable items={tableItemAlerts} isExpandable={false} cols={columns} pagination={{ itemsPerPage: 10 }} />
         )}
       </div>
     </div>
   );
 };
 
-function useColumns(): MatchedRulesTableColumnProps[] {
+function useColumns(): Array<DynamicTableColumnProps<AlertmanagerAlert>> {
+  const styles = useStyles2(getStyles);
+
   return [
     {
       id: 'state',
       label: 'State',
-      renderCell: function renderStateTag({ data: { matchedInstance } }) {
-        return <AlertStateTag state={matchedInstance.state} />;
+      renderCell: function renderStateTag({ data }) {
+        return <AmAlertStateTag state={data.status.state} />;
       },
-      size: '160px',
+      size: '120px',
+      className: styles.stateColumn,
     },
     {
       id: 'labels',
       label: 'Labels',
-      renderCell: function renderName({ data: { matchedInstance } }) {
-        return <AlertLabels labels={matchedInstance.labels} />;
+      renderCell: function renderName({ data }) {
+        return <AlertLabels labels={data.labels} className={styles.alertLabels} />;
       },
       size: 'auto',
     },
     {
       id: 'created',
       label: 'Created',
-      renderCell: function renderSummary({ data: { matchedInstance } }) {
-        return (
-          <>
-            {matchedInstance.activeAt.startsWith('0001')
-              ? '-'
-              : dateTime(matchedInstance.activeAt).format('YYYY-MM-DD HH:mm:ss')}
-          </>
-        );
+      renderCell: function renderSummary({ data }) {
+        return <>{data.startsAt.startsWith('0001') ? '-' : dateTime(data.startsAt).format('YYYY-MM-DD HH:mm:ss')}</>;
```

Comment:
There's the function `isNullDate` in `time.ts` to check for this in case that you'd like to use it here instead.

Reply:
Thanks! I'll use it